tests/test_utils.py

```diff
@@ -50,6 +51,7 @@ def test_with_env_pwd(self):
 
             git_hook_dir_path = get_git_hook_directory_path()
             self.assertEqual(git_hook_dir_path, git_dir_path / 'hooks')
+        os.environ['PWD'] = original_pwd
```

Comment:
Why is it required to change the PWD in the environment?

Reply:
I had testing issues (maybe race, not sure) with tests that didn't return the environ `PWD` to the original value.  I proactively (frowned upon maybe?) updated all test cases that changed the 'PWD' as part of the test.

Reply:
I don't understand why changing the environment should have some influence on this test. At least the code should get some comment why it needs to be changed.

Reply:
What OS did you run the tests, huh? I don't need that line in my code ...
What is the (error?/failure?) message you got?

Reply:
macOS.  I should have saved it.  I'll back it out and see if I can capture it again.

Reply:
Ok. It should not be the OS. At least my macOS Catalina works fine without that line.

Reply:
> I don't understand why changing the environment should have some influence on this test.
A test shouldn't have any side affect. In this case, the side affect is permentently c
 At least the code should get some comment why it needs to be changed.

Reply:
> I don't understand why changing the environment should have some influence on this test.
> At least the code should get some comment why it needs to be changed.

A test shouldn't have any side-effect; in this case, the global `os.environ` value for PWD. `autohooks` today doesn't use the PWD environment value (someone could put in an operational dependency on that in the future). But, dependencies might.

I took it out and couldn't reproduce it.  😞 I hope you can trust that it did fail on me simply for the reason that I don't know the codebase enough to just pick that could be a problem. It had to fail for me to wonder if that was side-effect could be avoided.

My implementation is a bit lame though.  Happy to take it out (vote: 👎 ) or vote 👍 for a context manager implementation:
```
    def test_with_env_pwd(self):
        with TemporaryDirectory() as f, mock.patch.dict(
            'os.environ', {'PWD': f}, clear=True
        ):
            temp_path = Path(f)
```

Reply:
> My implementation is a bit lame though. 

That implementation has been done by Björn and I am fine with it.

Reply:
If the error occurs again, just let us know.

Reply:
Will do.  

So, I want to thank you both for working with me to accommodate OS relics.  I really wanted a Python solution like `autohooks` and, it shouldn't be a surprise... I dove into the world of PR's for really the first _significant_ time with this pull.  Over this PR's duration, I have studied up on PR workflow with attention making it easier on reviewers.  Thank you very much for your patience.